src/WebJobs.Script/Host/FunctionAppValidationService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Diagnostics.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public sealed class FunctionAppValidationService : IHostedService
```

Comment:
Use `BackgroundService` and override `ExecuteAsync` method. The service will be disposed after `ExecuteAsync` returns. No need to keep the service after the folder check is done.